doc/domains.md

```diff
@@ -316,5 +316,162 @@ Provides classes specific for the United States of America:
 
 * **SSNGenerator**: Generates Social Security Numbers
 
-* **SSNValidator**: Validates Social Security Numbers and can be used as Benerator validator and as ConstraintValidator in Java Bean Validation (JSR
-  303)
\ No newline at end of file
+* **SSNValidator**: Validates Social Security Numbers and can be used as Benerator validator and as ConstraintValidator in Java Bean Validation (JSR 303)
+
+## faker domain
+
+The faker package provides the Generator class with library base on Java Faker (as well as Perl's Data::Faker library)
+
+* **FakerGenerator** : Generates data for many topics such as book, food, music... 
+
+Because this Generator has many topics, each topic has many properties, you have to choose topic and property and put it into 
+the 'generator' as parameters (like this `generator="new FakerGenerator('topic'','property')"`).
+
+Some topics have different data which base on the locale, you can change it by manual setting 'locale'.
+
+You can use the FakerGenerator like this:
+
+```xml
+<import domains="faker"/>
+
+<generate type="data" count="5" consumer="ConsoleExporter">
+    <attribute name="name" generator="new FakerGenerator('name','fullName')" locale="en_US"/>
+    <attribute name="HarryPotter" generator="new FakerGenerator('harryPotter','character')" locale="en"/>
+    <attribute name="dinner" generator="new FakerGenerator('food','dish')" locale="de_CH"/>
+</generate>
+```
+
+to get output similar to this:
+
+```
+data[name=Maira Lubowitz MD, HarryPotter=Pomona Sprout, dinner=French Fries with Sausages]
+data[name=Mr. Josue Kreiger, HarryPotter=Fang, dinner=Chicken Milanese]
+data[name=Henry Skiles, HarryPotter=Parvati Patil, dinner=Scotch Eggs]
+data[name=Minda Breitenberg, HarryPotter=Helga Hufflepuff, dinner=Lasagne]
+data[name=Nakita Schamberger, HarryPotter=Aragog, dinner=Poke]
+```
+
+### Supported topics:
+
+FakerGenerator can generate data for the following topics (with properties):
+
```

Comment:
please change in the way to integrate also the property type in a nice way , share with Huy , for example you take one half and huy the other half